### PR TITLE
Validate variable length relationships in CREATE and MERGE

### PR DESCRIPTION
**Pull request by nafraf**
Hi, this PR is to solve bug #2330

The functions ```_Validate_MATCH_Clauses()``` and ```_ValidateMergeRelation()``` were modified, in case of detect a variable length relationship, the error is reported.

Sample output:
```
127.0.0.1:6379> GRAPH.QUERY g "CREATE (a:A)-[e1:E1*]->(b:B)"
(error) Variable length relationships cannot be used in CREATE
127.0.0.1:6379> GRAPH.QUERY g "MERGE (a:A)-[e1:E1*]->(b:B)"
(error) Variable length relationships cannot be used in MERGE
```

### CODE REVIEW

**Comment by nafraf** [comment]
@AviAvni This PR was updated. Thanks

File: src/ast/ast_validations.c [hunk 1]
```diff
@@ -183,13 +183,20 @@ static AST_Validation _ValidateMultiHopTraversal
 	return AST_INVALID;
 }
 
-// Verify that MERGE doesn't redeclare bound relations, and that 
-// one reltype is specified for unbound relations (created).
+// Verify that MERGE doesn't redeclare bound relations, that one reltype is specified for unbound relations, 
+// and that there is not variable length relationship
```
Comment (raz-mon) [inline src/ast/ast_validations.c:187]: "and that the entity is not a variable length pattern"

**Review by raz-mon — COMMENTED** [review]
Hi @nafraf 
Looks great!
See one small comment.

File: src/ast/ast_validations.c (hunk 1 above)
Comment (nafraf) [reply src/ast/ast_validations.c]: Thanks, change done in [0344746](https://github.com/RedisGraph/RedisGraph/pull/2759/commits/03447464b4327f59d61f0a9ab5cd986bd424ed9f)

**Review by raz-mon — APPROVED** [review]

**Pull request merged**